src/pivotal-ui/components/forms/validator.js

```diff
@@ -61,5 +38,49 @@ var handleInput = function handleInput(e) {
       input.setCustomValidity('');
     }
   }
-  reflectValidity(input);
+  this.reflectValidity(input);
+};
+
+var ValidatedForm = function ValidatedForm(el) {
+  this.form = $(el);
+  this.inputs = this.form.find("input");
+  this.submitButton = this.form.find("[type='submit']");
+
+  this.form.attr("novalidate", "");
+
+  if(!this.form[0].checkValidity()){
+    this.submitButton.attr("disabled", "");
+  }
 };
+
+ValidatedForm.prototype.addListeners = function addListners() {
+  this.form.on("submit", handleSubmit.bind(this));
+  this.form.on("input", handleInput.bind(this));
+  this.form.on("click", "input[type='checkbox'], input[type='radio']", handleInput.bind(this));
+  this.form.on("focusout", handleBlur.bind(this));
+};
+
+ValidatedForm.prototype.reflectValidity = function reflectValidity(input){
+  if(!input.checkValidity()) {
+    removeError(input);
+    addError(input);
+  } else {
+    removeError(input);
+  }
+
+  if(this.form[0].checkValidity()){
+    this.submitButton.removeAttr("disabled");
+  } else {
+    this.submitButton.attr("disabled", "");
+  }
+};
+
+$(function(){
+  var forms = $("form[data-validate]");
+
+  $.each(forms, function(idx, form){
+    var vf = new ValidatedForm(form);
+    vf.addListeners();
```

Comment:
I'd like to separate all the concerns for testing so every line of the constructor method is testable.